### ISSUE TITLE
test: main branch test_stateless_cluster_linux not work

### ISSUE THREAD

**Issue by BohuTANG** [datafuselabs/databend#10222]
**Summary**

Two tests always failed

## test_stateless_cluster_linux
```
Having 3 errors! 51 tests passed.                     1 tests skipped.
The failure tests:
    /workspace/tests/suites/0_stateless/13_tpch/13_0009_q9.sql
    /workspace/tests/suites/0_stateless/13_tpch/13_0016_q16.sql
    /workspace/tests/suites/0_stateless/15_join_with_tpch_data/15_0007_right_anti_join.sql
Error: Process completed with exit code 1.
```
https://github.com/datafuselabs/databend/actions/runs/4270698458/jobs/7435087440#step:3:339

## test_stateful_cluster_linux
```
Having 2 errors! 52 tests passed.                     0 tests skipped.
The failure tests:
    /home/runner/work/databend/databend/tests/suites/1_stateful/04_mini_dataset/04_0000_mini_ontime.sh
    /home/runner/work/databend/databend/tests/suites/1_stateful/04_mini_dataset/04_0001_mini_hits.sh
Error: Process completed with exit code 1.
```
https://github.com/datafuselabs/databend/actions/runs/4270698458/jobs/7435086790#step:3:650

cc @zhang2014 @xudong963

**Comment by sundy-li** [comment]
Is that `find_subquery` should conside about the exchange plan ?  @xudong963

**Comment by xudong963** [comment]
> Is that `find_subquery` should consider about the exchange plan ? @xudong963

No, exchange plan is added after optmizer.

**Comment by zhang2014** [comment]
standalone hang; https://github.com/datafuselabs/databend/actions/runs/4281051489/jobs/7453788045

**Comment by zhang2014** [comment]
fixes in #10471
